frontend/lib/src/components/shared/BaseColorPicker/BaseColorPicker.tsx

```diff
@@ -47,113 +49,126 @@ export interface BaseColorPickerProps {
   help?: string
 }
 
-interface State {
-  /**
-   * The value specified by the user via the UI. If the user didn't touch this
-   * widget's UI, the default value is used.
-   */
-  value: string
-}
+const BaseColorPicker = (props: BaseColorPickerProps): React.ReactElement => {
+  const {
+    width,
+    showValue,
+    label,
+    labelVisibility,
+    help,
+    disabled,
+    value: propValue,
+  } = props
+  const [value, setValue] = React.useState(propValue)
+  const [popoverKey, setPopoverKey] = React.useState(0)
+  const theme: EmotionTheme = useTheme()
 
-class BaseColorPicker extends React.PureComponent<
-  BaseColorPickerProps,
-  State
-> {
-  public state: State = {
-    value: this.props.value,
-  }
+  // Reset the value when the prop value changes
+  React.useEffect(() => {
+    setValue(propValue)
+  }, [propValue])
 
-  public componentDidUpdate(prevProps: BaseColorPickerProps): void {
-    if (
-      prevProps.value !== this.props.value &&
-      this.props.value !== this.state.value
-    ) {
-      this.setState((_, prevProps) => {
-        return { value: prevProps.value }
-      })
+  React.useEffect(() => {
+    // 2021.06.30 - on Streamlit Sharing, ColorPicker throws a cross-origin
+    // error when its popover window is closed. There's an issue open in the
+    // react-color repo https://github.com/casesandberg/react-color/issues/806 -
+    // but it's months old and hasn't had a developer response.
+    const handleError = (error: ErrorEvent): void => {
+      if (error.error?.name === "SecurityError") {
+        logWarning(
+          `Swallowing ColorPicker SecurityError '${error.error.name}: ${error.error.message}'`
+        )
+        // We force an update by changing the key of the popover after this error,
+        // to re-mount the UIPopover - because the error sometimes cause it to be
+        // unmounted. This is an unfortunate hack.
+        setPopoverKey(prev => prev + 1)
+      }
     }
-  }
+
+    window.addEventListener("error", handleError)
+    return () => window.removeEventListener("error", handleError)
+  }, [])
 
   // Note: This is a "local" onChange handler used to update the color preview
   // (allowing the user to click and drag). this.props.onChange is only called
   // when the ColorPicker popover is closed.
-  private onColorChange = (color: ColorResult): void => {
-    this.setState({ value: color.hex })
+  const onColorChange = (color: ColorResult): void => {
+    setValue(color.hex)
   }
 
-  private onColorClose = (): void => {
-    this.props.onChange(this.state.value)
+  const onColorClose = (): void => {
+    props.onChange(value)
   }
```

Comment:
nit: I think we can put `onColorChange` and `onColorClose` into a useCallable to prevent reinitialization on every rendering.

Reply:
Done!